x-pack/plugins/security_solution/public/management/hooks/response_actions/use_send_execute_endpoint_request.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { UseMutationOptions, UseMutationResult } from '@tanstack/react-query';
+import type { IHttpFetchError } from '@kbn/core-http-browser';
+import { useMutation } from '@tanstack/react-query';
+import { KibanaServices } from '../../../common/lib/kibana';
+import { EXECUTE_ROUTE } from '../../../../common/endpoint/constants';
+import type { ResponseActionApiResponse } from '../../../../common/endpoint/types';
+import type { ExecuteActionRequestBody } from '../../../../common/endpoint/schema/actions';
+
+export const useSendExecuteEndpoint = (
+  options?: UseMutationOptions<ResponseActionApiResponse, IHttpFetchError, ExecuteActionRequestBody>
+): UseMutationResult<ResponseActionApiResponse, IHttpFetchError, ExecuteActionRequestBody> => {
+  return useMutation<ResponseActionApiResponse, IHttpFetchError, ExecuteActionRequestBody>(
+    (executeActionReqBody) => {
+      return KibanaServices.get().http.post<ResponseActionApiResponse>(EXECUTE_ROUTE, {
```

Comment:
Got curious, why we use here the `KibanaServices` and not the `useHttp` hook as we do in other components/hooks?

Reply:
We could use either `useHttp` or `KibanasServices.get`. They both return the same thing. I used `KibanaServices` similar to the other hooks in the folder for consistency.